### PR TITLE
fix: testing and relative imports fix

### PR DESCRIPTION
**Pull request by tadejsv**
Stuff to do

- [x] Import executor in `__init__.py`
- [x] Use only relative imports
- [x] Fix VGGish
- [x] Fix AudioCLIPEncoder
- [x] Add small load from config test to each executor
- [x] add CI to test docker image start @cristianmtr 
- [x] test manually that using relative imports and importing only `__init__.py` works for `jinahub://`

### CODE REVIEW

File: jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/model/clip/clip.py [hunk 1]
```diff
@@ -158,6 +158,9 @@ def patch_float(module):
 
         model.float()
 
+    global _tokenizer
```
Comment (JoanFM) [inline jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/model/clip/clip.py:161]: why do we touch this? let's keep it as it was

File: jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/utils/simple_tokenizer.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 @lru_cache()
 def default_bpe():
-    return os.path.join(os.path.dirname(os.path.abspath(__file__)), '..', 'assets', 'bpe_simple_vocab_16e6.txt.gz')
+    return os.path.join(os.getcwd(), '.cache', 'bpe_simple_vocab_16e6.txt.gz')
```
Comment (JoanFM) [inline jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/utils/simple_tokenizer.py:14]: Why do we rely on `cwd`? I prefer it the old way

**Review by JoanFM — CHANGES_REQUESTED** [review]
Why are we doing many more changes than the ones related to the actual fix?

File: jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py [hunk 3]
```diff
@@ -190,8 +190,7 @@ def tokenize(texts: Union[str, List[str]], context_length: int = 77) -> torch.Lo
 
     for i, tokens in enumerate(all_tokens):
         if len(tokens) > context_length:
-            leks = 'albaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalba'
-            raise RuntimeError(f"Input {leks} {len(texts[i])} {type(texts[i])} {texts[i]} is too long for context length {context_length}")
+            raise RuntimeError(f"Input {texts[i]} is too long for context length {context_length}")
```
Comment (JoanFM) [inline jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py:193]: do not change this code as it's not ours

File: jinahub/encoders/text/AudioCLIPTextEncoder/audio_clip/model/clip/clip.py [hunk 4]
```diff
@@ -197,8 +197,7 @@ def tokenize(texts: Union[str, List[str]], context_length: int = 77) -> torch.Lo
 
     for i, tokens in enumerate(all_tokens):
         if len(tokens) > context_length:
-            leks = 'albaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalbaalba'
-            raise RuntimeError(f"Input {leks} {len(texts[i])} {type(texts[i])} {texts[i]} is too long for context length {context_length}")
+            raise RuntimeError(f"Input {texts[i]} is too long for context length {context_length}")
```
Comment (JoanFM) [inline jinahub/encoders/text/AudioCLIPTextEncoder/audio_clip/model/clip/clip.py:200]: same here

Comment (tadejsv) [reply jinahub/encoders/text/AudioCLIPTextEncoder/audio_clip/model/clip/clip.py:200]: This was actually my change that I was using for debugging that unnoticaby made it's way to the main branch :')

File: jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/model/clip/clip.py (hunk 1 above)
Comment (cristianmtr) [reply jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/model/clip/clip.py:161]: @tadejsv

File: jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py (hunk 3 above)
Comment (cristianmtr) [reply jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py:193]: @tadejsv

File: jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/utils/simple_tokenizer.py (hunk 2 above)
Comment (tadejsv) [reply jinahub/encoders/audio/AudioCLIPEncoder/audio_clip/utils/simple_tokenizer.py:14]: I will change it to use `__file__`

File: jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py (hunk 3 above)
Comment (tadejsv) [reply jinahub/encoders/image/AudioCLIPImageEncoder/audio_clip/model/clip/clip.py:193]: Answered below

File: scripts/tests.sh [hunk 5]
```diff
@@ -33,8 +33,28 @@ if [[ -d "tests/" ]]; then
     python -m laserembeddings download-models
   fi
 
-  pytest -s -v tests/
+  python -m pytest -s -v tests/
```
Comment (tadejsv) [inline scripts/tests.sh:36]: Wait, do we need this still?

Comment (cristianmtr) [reply scripts/tests.sh:36]: no, my bad. Will revert

**Comment by tadejsv** [comment]
Ok, apart from catboost problems, this is almost done. Some things I would still do:
- `docker_args.txt` should be in `tests` folder
- I would split the docker test from the `tests.sh` script, and also create another step for it in `unit-tests` job.

File: jinahub/indexers/searcher/compound/FaissLMDBSearcher/requirements.txt [hunk 6]
```diff
@@ -1,3 +1,4 @@
-git+https://github.com/jina-ai/executors
-git+https://github.com/jina-ai/jina-commons
+jina_executors @ git+https://github.com/jina-ai/executors.git@main#egg=jinahub
+jina-commons @ git+https://github.com/jina-ai/jina-commons.git@v0.0.1#egg=jina-commons
```
Comment (tadejsv) [inline jinahub/indexers/searcher/compound/FaissLMDBSearcher/requirements.txt:2]:
```suggestion
jina-commons @ git+https://github.com/jina-ai/jina-commons.git@v0.0.2#egg=jina-commons
```

**Review by cristianmtr — APPROVED** [review]

File: jinahub/crafters/ImageNormalizer/tests/unit/test_normalizer.py [hunk 7]
```diff
@@ -1,19 +1,18 @@
...
-cur_dir = os.path.dirname(os.path.abspath(__file__))
+from ...normalizer import ImageNormalizer
 
 
 @pytest.fixture
 def numpy_image_uri(tmpdir):
     blob = np.random.randint(255, size=(96, 96, 3), dtype='uint8')
     im = fromarray(blob)
-    uri = os.path.join(tmpdir, 'tmp.png')
+    uri = str(Path(__file__).parent / 'tmp.png')
```
[the remark this answers is not in the document]
Comment (cristianmtr) [reply jinahub/crafters/ImageNormalizer/tests/unit/test_normalizer.py:15]: @tadejsv

File: jinahub/encoders/text/DPRTextEncoder/tests/unit/test_encoder.py [hunk 8]
```diff
@@ -23,7 +23,7 @@ def basic_encoder_ctx() -> DPRTextEncoder:
 
 
 def test_config():
-    encoder = BaseExecutor.load_config('../../config.yml')
+    encoder = Executor.load_config('../../config.yml')
```
Comment (JoanFM) [inline jinahub/encoders/text/DPRTextEncoder/tests/unit/test_encoder.py:26]: This one should have `__file__` or `cur_dir` as reference

File: jinahub/crafters/ImageNormalizer/tests/unit/test_normalizer.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (tadejsv) [reply jinahub/crafters/ImageNormalizer/tests/unit/test_normalizer.py]: fixed

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: jinahub/encoders/text/DPRTextEncoder/tests/unit/test_encoder.py (hunk 8 above)
Comment (tadejsv) [reply jinahub/encoders/text/DPRTextEncoder/tests/unit/test_encoder.py]: Good catch, thanks

**Review by cristianmtr — APPROVED** [review]

File: jinahub/encoders/text/DPRTextEncoder/Dockerfile [hunk 9]
```diff
@@ -1,4 +1,4 @@
-FROM jinaai/jina:2.0.12-perf
+FROM jinaai/jina:2.0-py38-perf
```
Comment (bwanglzu) [inline jinahub/encoders/text/DPRTextEncoder/Dockerfile:1]: what is this docker image, why sometimes it's py37 sometimes is py38?

**Review by bwanglzu — APPROVED** [review]
LGTM!

File: jinahub/encoders/text/DPRTextEncoder/Dockerfile (hunk 9 above)
Comment (tadejsv) [reply jinahub/encoders/text/DPRTextEncoder/Dockerfile:1]: Ok good point, when doing refactoring we will decide on a common docker image here

Comment (tadejsv) [reply jinahub/encoders/text/DPRTextEncoder/Dockerfile:1]: Now it's all over the place

Comment (cristianmtr) [reply jinahub/encoders/text/DPRTextEncoder/Dockerfile:1]: well this is a problem with any image you might choose to use. Needs to be fixed with dependabot, I'd say. out of scope on deciding / testing in this PR

**Pull request merged**